omod/src/main/java/org/openmrs/web/controller/LoginController.java

```diff
@@ -43,9 +43,10 @@ public String handleRequest(WebRequest webRequest, ModelMap model) {
 			}
 			webRequest.removeAttribute(WebConstants.INSUFFICIENT_PRIVILEGES, WebRequest.SCOPE_SESSION);
 		}
-		
-		//If there is a currently logged in user and they failed a privilege check, else go to login in page
-		if (Context.getAuthenticatedUser() != null && failedPrivilegeCheck) {
+		//If the user is logged in and there are no privilege problems, he should be redirected to home
+		if(Context.getAuthenticatedUser() != null && !failedPrivilegeCheck) {
+			return "/module/legacyui"+"/index";
```

Comment:
djazayeri: The ticket description asked to bring the user to the home page. So I thought this would be best. What page would you suggest we redirect him to? I'm not sure I understand the "forward:" refernce